### PR TITLE
Fix `contrib.metrics` issues

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Fixed the batches and sizes issues + edited integration tests + reduced computational load

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by KickItLikeShika** [comment]
I wanted to send a PR small number of metrics first to make the review easier if there is any edit. I can update this PR with the rest of the metric after the review or send another PR.

File: tests/ignite/contrib/metrics/test_average_precision.py [hunk 1]
```diff
@@ -87,18 +87,14 @@ def get_test_cases():
 
         test_cases = [
             (torch.randint(0, 2, size=(10,)).long(), torch.randint(0, 2, size=(10,)).long(), 1),
-            (torch.randint(0, 2, size=(100,)).long(), torch.randint(0, 2, size=(100,)).long(), 1),
             (torch.randint(0, 2, size=(10, 1)).long(), torch.randint(0, 2, size=(10, 1)).long(), 1),
-            (torch.randint(0, 2, size=(100, 1)).long(), torch.randint(0, 2, size=(100, 1)).long(), 1),
             # updated batches
             (torch.randint(0, 2, size=(16,)).long(), torch.randint(0, 2, size=(16,)).long(), 16),
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/test_average_precision.py:92]: Let's use size = 50 for batched test cases
```suggestion
            (torch.randint(0, 2, size=(50,)).long(), torch.randint(0, 2, size=(16,)).long(), 16),
```

File: tests/ignite/contrib/metrics/test_average_precision.py [hunk 2]
```diff
@@ -131,34 +127,30 @@ def get_test_cases():
         test_cases = [
             (torch.randint(0, 2, size=(10, 4)).long(), torch.randint(0, 2, size=(10, 4)).long(), 1),
             (torch.randint(0, 2, size=(50, 7)).long(), torch.randint(0, 2, size=(50, 7)).long(), 1),
-            (torch.randint(0, 2, size=(100, 4)).long(), torch.randint(0, 2, size=(100, 4)).long(), 1),
-            (torch.randint(0, 2, size=(200, 6)).long(), torch.randint(0, 2, size=(200, 6)).long(), 1),
             # updated batches
             (torch.randint(0, 2, size=(16, 4)).long(), torch.randint(0, 2, size=(16, 4)).long(), 16),
             (torch.randint(0, 2, size=(50, 7)).long(), torch.randint(0, 2, size=(50, 7)).long(), 16),
-            (torch.randint(0, 2, size=(100, 4)).long(), torch.randint(0, 2, size=(100, 4)).long(), 16),
-            (torch.randint(0, 2, size=(200, 6)).long(), torch.randint(0, 2, size=(200, 6)).long(), 16),
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/test_average_precision.py:140]: Same here, let's keep size=50 for batched test cases

**Review by vfdev-5 — COMMENTED** [review]
Thanks @KickItLikeShika , few comments on test size with batch

**Comment by vfdev-5** [comment]
Sounds good for multiple PRs

**Comment by vfdev-5** [comment]
@KickItLikeShika for other PR please make sure to have coherent test cases following all the comments from this and other PRs.

**Pull request merged**